src/xdp-utils.c

```diff
@@ -563,14 +563,65 @@ xdp_has_path_prefix (const char *str,
     }
 }
 
-void
-cleanup_temp_file (void *p)
+static gboolean
+ensure_sealed_memfd (int      fd,
```

Comment:
Not really a fan of this and the other function. Can we make this one `mapped_file_new_sealed_from_fd` and the other one `mapped_file_new_sealed_from_bytes`?

Reply:
Ah, no, that doesn't work because it represents an mmap'ed area. Then maybe a `XdpMemfd` gobject with `new_from_bytes`, `new_from_fd`, `seal`, `take_fd`, `dup_fd`.

Reply:
i added `XdpSealedFd`. I think it makes much more sense to only keep around a sealed fd and when ever an memfd is passed into a `XdpSealedFd` we ensure that it's sealed. We always want the fd to be immutable always anyways.

I didn't go with `XdpMemFd` since we don't care about the type of the fd as long as it is sealed. Right now, this is obviously only possible with fds created with `memfd_create()` by could change in future.